document-store/src/main/java/org/hypertrace/core/documentstore/parser/FromTypeExpressionVisitor.java

```diff
@@ -1,7 +1,15 @@
 package org.hypertrace.core.documentstore.parser;
 
+import org.hypertrace.core.documentstore.expression.impl.SubQueryJoinExpression;
 import org.hypertrace.core.documentstore.expression.impl.UnnestExpression;
 
 public interface FromTypeExpressionVisitor {
   <T> T visit(UnnestExpression unnestExpression);
+
+  /*
+   * Subquery join expression is not supported by default. Override this method to support it.
+   */
+  default <T> T visit(SubQueryJoinExpression subQueryJoinExpression) {
+    throw new UnsupportedOperationException("This operation is not supported");
```

Comment:
Typically, we should force all the implementors of visitor interface to implement all the methods.

Whether to throw exception or not should be the implementor's responsibility. 

We probably created the base class 
 `MongoSelectTypeExpressionParser` for a similar purpose (without having a default implementation in the interface).

The reason being, it may make sense to implement this in other implementations (other than MongoDB).

Reply:
updated it